tests/python/test_dataset_conversion.py

```diff
@@ -0,0 +1,610 @@
+        DatasetConverter.add_metadata_for_xarray(
+            ds2, 
+            data_names=1,
+            dim_names=dimname,
+            coord_names=coordname, 
+            attr_names=attrname
+        )
+    with pytest.raises(TypeError):
+        DatasetConverter.add_metadata_for_xarray(
+                ds2, 
+                data_names=dataname,
+                dim_names=[1,2],
+                coord_names=coordname, 
+                attr_names=attrname
+            )
+    with pytest.raises(TypeError):   
+        DatasetConverter.add_metadata_for_xarray(
+                ds2, 
+                data_names=dataname,
+                dim_names=dimname,
+                coord_names=[3,4], 
+                attr_names=attrname
+            )
+    with pytest.raises(TypeError):    
+        DatasetConverter.add_metadata_for_xarray(
+                ds2, 
+                data_names=dataname,
+                dim_names=dimname,
+                coord_names=coordname, 
+                attr_names=[5,6,7]
+            )
+
+#------- beginning of 1d transform_to_xarray tests-------
+
+def test_transform_to_xarray_1d():
+    """Test transform_to_xarray method with correct 1d
+    tensor data and assert equality
+    """
+    
+    ds1 = Dataset("ds-1d")
+    add_1d_data(ds1)
+    # Calling method add_metadata_for_xarray on 1D dataset 
+    # good data and prerequisite for transform_to_xarray 
+    DatasetConverter.add_metadata_for_xarray(
+        ds1, 
+        data_names=["1ddata"],
+        dim_names=["dim_data_x"],
+        coord_names=["x"],
+        attr_names=["units","longname","convention"]
+    )
+    # Calling method add_metadata_for_xarray for longitude coordinate
+    DatasetConverter.add_metadata_for_xarray(
+        ds1,
+        data_names=["x"], 
+        dim_names=["dim_data_x"],
+        attr_names=["x_coord_units","x_coord_longname"] 
+    )
+    # Compare generated Xarray from 1D data to initial Xarray
+    d1_xarray_ret = DatasetConverter.transform_to_xarray(ds1)
+    d1_transformed = d1_xarray_ret["1ddata"]
+    assert(ds_1d.identical(d1_transformed))
+
+def test_bad_data_transform_to_xarray_1d(): 
+    """Test transform_to_xarray method with incorrect 1d data
+    """
+    
+    ds1 = Dataset("ds-1d")
+    add_1d_data(ds1)
+    dataname = ["1ddata"]
+    dimname = ["dim_data_x"]
+    coordname = ["x"]
+    attrname = ["units","longname","convention"]
+    cdataname = ["x"]
+    cdimname = ["dim_data_x"]
+    cattrname = ["x_coord_units","x_coord_longname"] 
+   
+    # Calling method add_metadata_for_xarray on 1D dataset 
+    # good data and prerequisite for transform_to_xarray 
+    with pytest.raises(RedisKeyError): 
+        DatasetConverter.add_metadata_for_xarray(
+            ds1, 
+            data_names=["baddata"],
+            dim_names=dimname,
+            coord_names=coordname,
+            attr_names=attrname
+        )
+        # Calling method add_metadata_for_xarray for x coordinate
```

Comment:
If you get a key error on the first call, I don't think the second call will even happen. If you're expecting the error on the second call, please pull the first out of the with clause.

Reply:
These tests are not testing the add_metadata_for_xarray methods, they are testing the transform to xarray methods. However, because add_metadata_for_xarray is a prerequisite for the transform method, that is where the bad input would be coming in if that makes sense

Reply:
never mind, I figured it out with andrew